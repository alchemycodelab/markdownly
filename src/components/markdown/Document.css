.Document {
  display: flex;
  flex-direction: column;
}
