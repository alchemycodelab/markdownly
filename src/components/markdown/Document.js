import React from 'react';
import Preview from './Preview';
import Editor from './Editor';
import styles from './Document.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';



const Document = ({ markdown, updateMarkdown }) => {

  return (
    <>
      <div className={styles.Document}>
        <Editor markdown={markdown} updateMarkdown={updateMarkdown} />
        <Preview markdown={markdown} />
      </div>
    </>
  );
};

const mapStateToProps = state => ({
  markdown: getMarkdown(state)
});

const mapDispatchToProps = dispatch => ({
  updateMarkdown({ target }) {
    dispatch({ type: 'UPDATE_MARKDOWN', payload: target.value });
  }
});

Document.propTypes = {
  markdown: PropTypes.string.isRequired,
  updateMarkdown: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Document);
