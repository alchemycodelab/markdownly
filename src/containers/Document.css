.Document {
  display: flex;
  padding-top: 2em;
}
