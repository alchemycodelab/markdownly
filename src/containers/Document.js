import React from 'react';
import Preview from '../components/markdown/Preview';
import Editor from '../components/markdown/Editor';
import styles from './Document.css';
import { useSelector, useDispatch } from 'react-redux';
import { getMarkdown } from '../selectors/documentSelectors';
import { updateMarkdown } from '../actions/documentActions';


export default function Document() {
  const markdown = useSelector(state => getMarkdown(state));
  const dispatch = useDispatch();
  const letUpdateMarkdown = ({ target }) => dispatch(updateMarkdown(target.value));
  return (
    <>
      <div className={styles.Document}>
        <Editor markdown={markdown} updateMarkdown={letUpdateMarkdown} />
        <Preview markdown={markdown} />
      </div>
    </>
  );
}


