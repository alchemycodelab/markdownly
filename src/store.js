import { createStore, compose } from 'redux';
import reducer from './reducers';
import { saveState } from './localStorage';


const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducer,
  composeEnhancers()
);

store.subscribe(() => {
  const state = store.getState();
  saveState({ state: state });
});

export default store;
