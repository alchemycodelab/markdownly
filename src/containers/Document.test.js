import React from 'react';
import { shallow } from 'enzyme';
import Document from './Document';
import store from '../store';

describe('Document component', () => {
  it('renders Document', () => {
    const wrapper = shallow(<Document store={store}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
