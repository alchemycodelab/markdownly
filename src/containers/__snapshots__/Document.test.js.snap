// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document component renders Document 1`] = `
<ContextProvider
  value={null}
>
  <Document
    markdown="# hi there"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    updateMarkdown={[Function]}
  />
</ContextProvider>
`;
